samza-test/src/test/java/org/apache/samza/test/operator/StreamApplicationIntegrationTestHarness.java

```diff
@@ -247,7 +254,7 @@ public void runApplication(StreamApplication streamApplication, String appName,
     }
 
     app = streamApplication;
-    runner = ApplicationRunner.fromConfig(new MapConfig(configs));
+    runner = (RemoteApplicationRunner) ApplicationRunner.fromConfig(new MapConfig(configs));
```

Comment:
Due to the existing limit in Samza test suite, we have to use `RemoteApplicationRunner` instead of `LocalApplicationRunner`. Prateek left comment in `StreamApplicationIntegrationTestHarness` that says "We can't use LocalApplicationRunner since ZkJobCoordinator doesn't currently create". I think Prateek can explain more deeply why we have to use `RemoteApplicationRunner`.

Reply:
HI, @lindong28 thanks for the response. I should have made my point more explicit. The comment here is not whether the current integration test is actually using which ApplicationRunner class via configuration. The request here is not limiting the ApplicationRunner implementation class that could be used in this test harness.

Reply:
Thanks @nickpan47. I have updated the patch to move getIntermediateStreams() into AbstractApplicationRunner. Thus all AbstractApplicationRunner can be used in the test harness.